### ISSUE TITLE
Inconsistent results from cause filter in RECAP Search

### ISSUE THREAD

**Issue by albertisfu** [freelawproject/courtlistener#3993]
We received a report about a user not being able to get the expected results from the `cause` filter.

https://www.courtlistener.com/?q=&type=r&order_by=score%20desc&cause=%2231%3A3730%20Qui%20Tam%20False%20Claims%20Act%22

The query doesn't return any results.

Filtering using a fielded search does work.
https://www.courtlistener.com/?q=cause%3A%2231%3A3730%20Qui%20Tam%20False%20Claims%20Act%22&type=r&order_by=score%20desc

So the problem might be related to the type of query used for the filter, where some terms break the query.

A partial query, removing terms from original query in the filter, also works:
https://www.courtlistener.com/?q=&type=r&order_by=score%20desc&cause=Qui%20Tam%20False%20Claims

**Comment by mlissner** [comment]
Looks like the colon might be to blame. Since we have a workaround, I think it's OK to continue on your API work and then get back to this during a cleanup round.

**Comment by ERosendo** [comment]
I've been troubleshooting this issue, and I think the faceted search can find matches because we're using `quote_field_suffix` in full-text queries. This tells Elasticsearch to look for quoted words in a specific field (in this case, `cause.exact`).

I tried adjusting the **cause** filter to use the `.exact` version of the field, and the first two queries from Alberto produced identical results.

Solution Options:

1. **Disable Stemming for cause Field:** We can tweak the `cause` filter to always use the **exact** version. This will disable stemming for this field.

2. **Enhance build_text_filter Function:** Alternatively, we can adjust the `build_text_filter` function to use the `quote_field_suffix` feature, similar to how it's used in full-text queries. This approach offers greater flexibility, allowing users to mix exact and stemmed searches by using quotation marks.


@albertisfu @mlissner I'd appreciate your feedback on this.

**Comment by mlissner** [comment]
I agree. In general, users should get the same functionality from the faceted filters that they do from the main query box.

**Comment by ERosendo** [comment]
> does the bug you found create a bug in our system?

We're not sure. We were investigating the potential impact of the bug yesterday. While the documentation suggests it should produce tokens with no positionLength, we've observed unexpected multi-position tokens in the output when processing strings from the `synonyms_en.txt` file.

During my review of the [Flatten Graph](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-flatten-graph-tokenfilter.html) filter, I came across a section in the [token graph documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/token-graphs.html) that warns about certain filters, like `synonym` and `word_delimiter`, producing [invalid token graphs](https://www.elastic.co/guide/en/elasticsearch/reference/current/token-graphs.html#token-graphs-invalid-token-graphs), which can negatively impact search. this didn't seem to directly affect our code since most tests were passing.  However, the example invalid graph and the flatten graph output appeared identical. Intrigued by this inconsistency, I delved deeper to ensure we aren't using invalid graphs for search and discovered the issue I reported. 

Here's a screenshot that compares the graphs from the documentation side-by-side.

![Image](https://github.com/user-attachments/assets/a15f45be-ce02-4881-bbb8-a090ce0c953d)

I've opened the issue on the ES repo to get the Elasticsearch team's input on this filter. I believe they can help us understand how it works and determine if the behavior we saw is a bug or not. Hopefully, they can also provide a clearer example, which will help us accurately assess whether it could potentially cause invalid graphs.

**Comment by mlissner** [comment]
Sounds good. I really hoped Elastic would do better with synonyms, but it seems like it uses the same sorts of hacks as Solr. Bummer.

**Comment by mlissner** [comment]
I think we concluded there's nothing we can do here, right? Shall we set this to Done and get it off our sprint?

**Comment by albertisfu** [comment]
Well, the original issue reported by the user using a phrase search is actually solved by https://github.com/freelawproject/courtlistener/pull/4814 so we can merge it and mark this as done.

The remaining identified issue related to non-phrase searching and synonyms is currently on hold, as discussed in:
https://github.com/freelawproject/courtlistener/issues/4816
